OpenCV Error: Assertion failed (vc_.isOpened()) in reset

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)

- OpenCV => 3.3.1
- Operating System / Platform => Ubunutu Mate 64Bit
- Compiler => CMake



##### Detailed description

I compiled [super_resolution.cpp](https://github.com/opencv/opencv/blob/master/samples/gpu/super_resolution.cpp), when I run it I issue in opening video file. 

Command: 
`./super_resolution vid.mp4`

Error message: 
`OpenCV Error: Assertion failed (vc_.isOpened()) in reset, file /home/hat/Downloads/opencv-3.3.1/modules/superres/src/frame_source.cpp, line 153
terminate called after throwing an instance of 'cv::Exception'
  what():  /home/hat/Downloads/opencv-3.3.1/modules/superres/src/frame_source.cpp:153: error: (-215) vc_.isOpened() in function reset`

CMakeLists.txt:
`
cmake_minimum_required(VERSION 2.8)
project( super_resolution ) # change this 
find_package( OpenCV REQUIRED )
include_directories( ${OpenCV_INCLUDE_DIRS} )
add_executable( super_resolution super_resolution.cpp ) # change this
target_link_libraries( super_resolution ${OpenCV_LIBS} )
`



<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->

--- a/samples/gpu/super_resolution.cpp
+++ b/samples/gpu/super_resolution.cpp
@@ -55,7 +55,7 @@ static Ptr<cv::superres::DenseOpticalFlowExt> createOptFlow(const string& name, 
 int main(int argc, const char* argv[])
 {
     CommandLineParser cmd(argc, argv,
-        "{ v video      |           | Input video }"
+        "{ v video      |           | Input video (mandatory)}"
         "{ o output     |           | Output video }"
         "{ s scale      | 4         | Scale factor }"
         "{ i iterations | 180       | Iteration count }"
@@ -65,14 +65,14 @@ int main(int argc, const char* argv[])
         "{ h help       | false     | Print help message }"
     );
 
-    if (cmd.get<bool>("help"))
+    const string inputVideoName = cmd.get<string>("video");
+    if (cmd.get<bool>("help") || inputVideoName.empty())
     {
         cout << "This sample demonstrates Super Resolution algorithms for video sequence" << endl;
         cmd.printMessage();
         return EXIT_SUCCESS;
     }
 
-    const string inputVideoName = cmd.get<string>("video");
     const string outputVideoName = cmd.get<string>("output");
     const int scale = cmd.get<int>("scale");
     const int iterations = cmd.get<int>("iterations");

